www/js/diary/list/DateSelect.tsx

```diff
@@ -4,88 +4,86 @@
     when we have fully migrated to React Native.
 */
 
-import React, { useEffect, useState } from "react";
-import { angularize } from "../../angular-react-helper";
-import { func, object } from "prop-types";
+import React, { useEffect, useState, useMemo, useContext } from "react";
+import { View, Text } from "react-native";
 import moment from "moment";
+import color from "color";
+import { LabelTabContext } from "../LabelTab";
+import { DatePickerModal } from "react-native-paper-dates";
+import { Divider, Button, IconButton, useTheme } from "react-native-paper";
+import i18next from "i18next";
+import { useTranslation } from "react-i18next";
 
 const DateSelect = ({ tsRange, loadSpecificWeekFn }) => {
 
-  const [dateRange, setDateRange] = useState('-\n-');
-  const [selDate, setSelDate] = useState('-');
+  const { pipelineRange } = useContext(LabelTabContext);
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const [open, setOpen] = React.useState(false);
+  const [dateRange, setDateRange] = useState(['-', '-']);
+  const [selDate, setSelDate] = useState(null);
+  const minMaxDates = useMemo(() => ({
+    startDate: new Date(pipelineRange?.start_ts * 1000),
+    endDate: new Date(pipelineRange?.end_ts * 1000),
+  }), [pipelineRange]);
 
   useEffect(() => {
     if (!tsRange.oldestTs) return;
-    const start = moment.unix(tsRange.oldestTs).format('YYYY-MM-DD');
-    const end = moment.unix(tsRange.latestTs).format('YYYY-MM-DD');
-    setDateRange(start + '\n' + end);
-    const mid = moment.unix((tsRange.oldestTs + tsRange.latestTs) / 2).format('YYYY-MM-DD');
-    setSelDate(mid);
+    const displayStartTs = Math.max(tsRange.oldestTs, pipelineRange.start_ts);
```

Comment:
Yes we use `pipelineRange` here, but I don't think we need to watch it for changes.
We aren't expecting `pipelineRange` to change during a session right?

Reply:
true, it will not change on the phone during a session. Note that the pipeline can be running in parallel on the server, but we won't see the range updated unless we contact the server again.